test/unit/cssTest.js

```diff
@@ -0,0 +1,268 @@
+      css: {
+        compiler: {
+          nodeModule: 'roosevelt-less',
+          params: {
+            cleanCSS: {
+              advanced: true,
+              aggressiveMerging: true
+            },
+            sourceMap: null
+          }
+        }
+      },
+      generateFolderStructure: true
+    }, 'initServer')
+
+    // fork the app
+    const testApp = fork(path.join(appDir, 'app.js'), ['-p'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // on the message being sent back from the initialized app, test to see that they are there
+    testApp.on('message', () => {
+      // look for the folder
+      let pathToCompiledFolder = path.join(appDir, 'statics', '.build', 'css')
+      let cssCompiledArray = klawsync(pathToCompiledFolder)
+      // look at each file and see that it checks out with the test array
+      cssCompiledArray.forEach((file) => {
+        let test = pathOfCSSCompiledfilesArray.includes(file.path)
+        assert.equal(test, true)
+      })
+      // kill the app and say the test is done afterwards
+      testApp.kill()
+      done()
+    })
+  })
+
+  it('should only compiled the files that are whitelisted', function (done) {
+    // generate the static files and folder
+    generateStaticFolder()
+
+    // make a test array for whitelisted files
+    const pathOfWhiteListedArray = [
+      path.join(appDir, 'statics', '.build', 'css', 'b.css'),
+      path.join(appDir, 'statics', '.build', 'css', 'c.css')
+    ]
+
+    // generate the app
+    generateTestApp({
+      appDir: appDir,
+      css: {
+        compiler: {
+          nodeModule: 'roosevelt-less',
+          params: {
+            cleanCSS: {
+              advanced: true,
+              aggressiveMerging: true
+            },
+            sourceMap: null
+          }
+        },
+        whitelist: ['b.css', 'c.css']
+      },
+      generateFolderStructure: true
+    }, 'initServer')
+
+    // fork the app
+    const testApp = fork(path.join(appDir, 'app.js'), ['-p'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // on the message being sent back from the initialized app, test to see that the whitelist files were compiled
+    testApp.on('message', () => {
+      // look for the folder
+      let pathToCompiledFolder = path.join(appDir, 'statics', '.build', 'css')
+      let cssCompiledArray = klawsync(pathToCompiledFolder)
+      // look at each file and see that it checks out with the test array
+      cssCompiledArray.forEach((file) => {
+        let test = pathOfWhiteListedArray.includes(file.path)
+        assert.equal(test, true)
+      })
+      // kill the app and say the test is done afterwards
+      testApp.kill()
+      done()
+    })
+  })
+
+  it('should rename the compiled output file to what is on the output parameter', function (done) {
+    // Generate the static files and folder
+    generateStaticFolder()
+
+    // change the compiled path array to reflect the changed file name
+    pathOfCSSCompiledfilesArray = [
```

Comment:
We should avoid changing the values of variables inside of tests, instead create a new one in here specific to this test.